File: spring-boot-rest-api/src/main/java/br/com/uabrestingaseca/biblioteca/model/Reserva.java
package br.com.uabrestingaseca.biblioteca.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Objects;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "reserva")
public class Reserva implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "num_registro", referencedColumnName = "num_registro")
    @JsonInclude(value = JsonInclude.Include.NON_NULL)
    private Exemplar exemplar;

    @Transient
    @JsonInclude(value = JsonInclude.Include.NON_NULL)
    private Integer exemplarNumRegistro = null;

    @ManyToOne
    @NotNull(message = "Usuário deve ser informado")
    private Usuario usuario;

    @Transient
    @JsonInclude(value = JsonInclude.Include.NON_NULL)
    private Integer usuarioId = null;

    @Column(name = "data_hora")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dataHora;

    @Column(name = "data_limite")
    private LocalDate dataLimite;

    @Column(name = "data_hora_retirada")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dataHoraRetirada;

    private boolean ativa = true;

    @Deprecated
    public Reserva(Integer id) {
        this.id = id;
    }
}

File: biblioteca-react/src/model/Reserva.ts
import DateTimeUtil from "../util/DateTimeUtil";
import { Exemplar, Usuario } from "./";

export class Reserva{

    constructor(
        public id?: number,
        public exemplar?: Exemplar,
        public usuario?: Usuario,
        public dataHora?: Date,
        public dataLimite?: Date,
        public dataHoraRetirada?: Date,
        public ativa?: boolean
    ){};

    public static createFromData(data : any) : Reserva{
        const reserva = new Reserva();
        reserva.id = data.id;
        reserva.exemplar = Exemplar.createFromData(data.exemplar);
        reserva.usuario = Usuario.createFromData(data.usuario);
        reserva.dataHora = DateTimeUtil.fromAPIDateTime(data.dataHora);
        reserva.dataLimite = DateTimeUtil.fromAPIDate(data.dataLimite);
        reserva.dataHoraRetirada = DateTimeUtil.fromAPIDateTime(data.dataHoraRetirada);
        reserva.ativa = data.ativa;
        return reserva;
    }

}